docs/frontend-system/architecture/05-extension-overrides.md

```diff
@@ -10,38 +10,107 @@ description: Frontend extension overrides
 
 ## Introduction
 
-<!--
+An extension override is another building block in the Frontend System that allows you to programmatically override app or plugin extensions. Extension overrides are recommended in the following cases:
 
-Introduce extension overrides and how they can be shipped in separate packages but also internally within a project too.
+- Override the app skeleton layout;
+- Override a default api implementation;
+- Override a default app component;
+- Override a plugin's default page;
+- Etc.
 
-Talk about how extension overrides
+It is also important to clarify when an override isn't needed. Creating an extension override is not recommended when:
 
--->
+- When you just want to configure a plugin, such as changing the attachment point, a path or any other setting. Be sure to read the extension documentation before overriding an extension to make sure you cannot configure it to work the way you want;
+- It is already possible to enable an extension that is already provided by the plugin. Imagine we have a Search plugin that provides a default search result item extension for rendering search results. Consider also that we have a Catalog plugin with a disabled Catalog search result item extension. To change the way Catalog search result items are rendered, you don't need to override the default result item extension. The only thing you need to do is enable the Catalog search result extension via the config file. For other types of search results, the default Search result item will continue to be used.
```

Comment:
Feeling that this is perhaps a little it contrived 😅, isn't it basically "double check that there isn't already an extensions that does what you want that you could enable instead"?

Reply:
made into regular paragraph text